src/datasources/perf-ds/modal_ctrl.js

```diff
@@ -36,6 +50,24 @@ class ModalCtrl {
     }
   }
 
+  nextPage(){
+    if((this.currentPage + 1) == this.$scope.ctrl.numberOfPages){
+      return;
+    }
+    this.offset += 25;
+    this.currentPage++;
+    this.searchForRows();
```

Comment:
In the case where the given search function does not support the offset and always returns the full results, this will cause all of the data to be fetched again - which we don't want.

Reply:
Thanks for mentioning! Client-side pagination added: https://github.com/OpenNMS/opennms-helm/pull/257/commits/9dbc78d3ba0977c483cc565571e965576ced4416